### PR TITLE
Refine line comment regexp

### PR DESCRIPTION
**Pull request by ulidtko**
Before:

After:

### CODE REVIEW

File: grammars/rpm-spec.cson [hunk 1]
```diff
@@ -505,7 +505,7 @@
   'comments':
     'patterns': [
       {
-        'match': '#.*$',
+        'match': '^\s*#.*$',
```
Comment (waveclaw) [inline grammars/rpm-spec.cson:508]: This is a good issue to bring up.  The Microsoft sharp languages invalidate the assumptions made by trivial parsers for languages with an octothorpe for the start-of-line-comment character.  (Microsoft marketing spawned names like .Net also cause parsing problems for other cases, but that is a digression.)

The offered solution is not one that I can accept. This expression means that only lines that start with comments or whitespace plus comments are comments.

In other words, this excludes valid end-of-line comments from appearing anywhere in the file.

I have a partial solution in the work.  However this requires either unclear and clever tricks or working negative negative-width assertions for simplicity.  In any case the required changes will be large enough to force a 0.9.0 version at least.

**Comment by waveclaw** [comment]
Please see if the explicit list for the Microsoft sharp languages helps with your use case.

There is still an outstanding wart that could use improvement upon.  The current regular expression lexer in the parser does not consider word boundaries before the language tag.  This means that CC# or FMFMFM# will not start a comment in version 0.9.0.
